### PR TITLE
[7.4-only] LPS-122449 Replace usages of dom.delegate in Item Selector, Roles, and Site Navigation

### PR DESCRIPTION
**Pull request by kresimir-coko**
This PR removes usages of `dom.delegate` from files found inside the `Item Selector`, `Roles` and `Site Navigation` modules and replaces those usages with the modern `delegate` utility.

Continuation of #531

### CODE REVIEW

**Comment by kresimir-coko** [comment]
ci:test:relevant

**Comment by wincent** [comment]
ci:test:sf

File: modules/apps/site-navigation/site-navigation-admin-web/src/main/resources/META-INF/resources/view.jsp [hunk 1]
```diff
@@ -147,8 +147,10 @@ SiteNavigationAdminManagementToolbarDisplayContext siteNavigationAdminManagement
 	</liferay-ui:search-container>
 </aui:form>
 
-<aui:script require="metal-dom/src/all/dom as dom,frontend-js-web/liferay/modal/commands/OpenSimpleInputModal.es as openSimpleInputModal" sandbox="<%= true %>">
-	var renameSiteNavigationMenuClickHandler = dom.delegate(
+<aui:script require="frontend-js-web/liferay/modal/commands/OpenSimpleInputModal.es as openSimpleInputModal,frontend-js-web/liferay/delegate/delegate.es as delegateModule" sandbox="<%= true %>">
```
Comment (wincent) [inline modules/apps/site-navigation/site-navigation-admin-web/src/main/resources/META-INF/resources/view.jsp:150]: We can try forwarding this as-is, @kresimir-coko, but there is a non-zero chance that Brian might ask this to be sorted, so FWIW I think you may as well sort it (`delegate`/`delegateModule` comes before `modal`/`openSimpleInputModal`).

**Review by wincent — APPROVED** [review]
lg2m but before running `ci:forward` would you mind swapping the order of the module imports in the place that I commented @kresimir-coko?

File: modules/apps/site-navigation/site-navigation-admin-web/src/main/resources/META-INF/resources/view.jsp (hunk 1 above)
Comment (kresimir-coko) [reply modules/apps/site-navigation/site-navigation-admin-web/src/main/resources/META-INF/resources/view.jsp:150]: @wincent Sure, I wasn't sure what the rule here was, I'll push and ping you when it's ready.

Comment (kresimir-coko) [reply modules/apps/site-navigation/site-navigation-admin-web/src/main/resources/META-INF/resources/view.jsp]: @wincent I've pushed a commit that sorts it.

Comment (wincent) [reply modules/apps/site-navigation/site-navigation-admin-web/src/main/resources/META-INF/resources/view.jsp]: > I wasn't sure what the rule here was

I don't know if there is a *rule* as such (and we probably don't import multiple modules with `require` all that often), but this just seems like the kind of thing Brian would ask about.

better-safe-than-sorry

Comment (wincent) [reply modules/apps/site-navigation/site-navigation-admin-web/src/main/resources/META-INF/resources/view.jsp]: @kresimir-coko: Can I be really annoying and ask you to drop the `sandbox` attribute as well? I don't think it's required with, er, `require`, as [I commented on another PR](https://github.com/liferay-frontend/liferay-portal/pull/558#discussion_r531502056).

Once you've done that, feel free to hit `ci:forward` and we'll see if the CI smiles upon you.

Comment (kresimir-coko) [reply modules/apps/site-navigation/site-navigation-admin-web/src/main/resources/META-INF/resources/view.jsp]: Sure @wincent , I hate it anyway, IDK what it does (I knew at one point) and I just don't like it for some reason, so I'll gladly terminate it.

**Comment by kresimir-coko** [comment]
ci:forward

**Comment by kresimir-coko** [comment]
ci:forward

**Comment by wincent** [comment]
That failure looks to be pretty consistent, so the forward is probably not going to work. @kresimir-coko: if you look at the steps in the failing Poshi test, are you able to reproduce the problem? It might be a real bug, although I can't see anything obviously wrong with the code. Maybe @PablitoBonito or @john-co can verify that the issue is legit.

**Comment by wincent** [comment]
ci:test:bundle

**Comment by kresimir-coko** [comment]
@wincent 
> if you look at the steps in the failing Poshi test, are you able to reproduce the problem? 

I'll take a look.

**Comment by kresimir-coko** [comment]
@wincent @PablitoBonito @john-co 
So from my understanding, looking at the Poshi report I get to the correct page, click `Save` and then a dismissable alert pops up, Poshi reports that the dismissing the alert doesn't work, but in my attempts to reproduce it, **the alert gets dismissed without issue.**

**Comment by kresimir-coko** [comment]
ci:test:relevant

**Comment by jbalsas** [comment]
Let's just try to get this over with... I have similar concerns to the ones in https://github.com/liferay-frontend/liferay-portal/pull/557#pullrequestreview-540870900, but seeing as we have a tight window, let's see if we can get these 2 merged.

**Comment by jbalsas** [comment]
ci:forward

**Pull request closed without merging**